graphbolt/src/cuda/unique_and_compact_impl.cu

```diff
@@ -299,10 +293,25 @@ UniqueAndCompactBatched(
   // Utilizes a sort based algorithm, the mapped id of a vertex part of the
   // src_ids but not part of the unique_dst_ids will be monotonically increasing
   // as the actual vertex id increases. Thus, it is deterministic.
-  return UniqueAndCompactBatchedSortBased(src_ids, dst_ids, unique_dst_ids);
+  auto results3 =
```

Comment:
Why results3 & results4?

Reply:
Because the `world_size>1` code path is only supported for the Map based implementation. So the 4th argument is not returned by the sort based implementation. So we modify the output to fill the 4th output ourselves.